src/mongoose.ts

```diff
@@ -145,13 +145,10 @@ export class MongoosePlugin extends BasePlugin<typeof mongoose> {
     return (originalThen: Function) => {
       return function patchedThen(this: any) {
         if(this._otContext) {
-          return thisPlugin._tracer.withSpan(this._otContext, () => {
-            return originalThen.apply(this, arguments);
-          });  
-        }
-        else {
-          return originalThen.apply(this, arguments);
+          return thisPlugin._tracer.withSpan(this._otContext, () => originalThen.apply(this, arguments))
```

Comment:
The PR is linked to issue #35 , but it looks like it still uses `withSpan` instead of `bind`.